### PR TITLE
Added RBAC for CRD hpereplicationdeviceinfos

### PR DESCRIPTION
**Pull request by wdurairaj**
We plan to use the custom resource definition for the storing the replicated device information as part of supporting the 3PAR/Primera CSP 1.1 release which adds the capability of peer persistence. 

For this , we need a CRD which will allow us to store details (name/value pairs) for individual array in replicated configuration like

```yaml
apiVersion: storage.hpe.com/v1
kind: HPEReplicationDeviceInfo
metadata:
  name: replication-crd2
spec:
  target_array_details:
  - target_cpg: FC_r6
    target_name: CSSOS-SSA05
    target_secret: secret-r
    target_snap_cpg: FC_r6
    target_secret_namespace: kube-system
```

Signed-off-by: William Durairaj <w_durairaj@yahoo.com>

### CODE REVIEW

**Comment by wdurairaj** [comment]
> looks good. Can you also add RBAC under operators/hpe-csi-driver/deploy/role.yaml and deploy/olm-catalog/1.3.0/hpe-csi-operator*.yaml as well?
@shivamerla 
I have added these files already , can you confirm ?

 ...-catalog/hpe-csi-driver-operator/1.3.0/hpe-csi-operator.v1.3.0.clusterserviceversion.yaml 
 operators/hpe-csi-operator/deploy/role.yaml

**Comment by sneharai4** [comment]
@wdurairaj Do we need to add mode as part of the crd

File: helm/charts/hpe-csi-driver/crds/hpe-replicated-device-info-crd.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                    targetSecret:
+                      description: Secret of the replicated array 
+                      type: string 
+          
+          type: array
+  version: v1
+status:
+  acceptedNames:
+    kind: ""
+    plural: ""
```
Comment (sneharai4) [inline helm/charts/hpe-csi-driver/crds/hpe-replicated-device-info-crd.yaml:42]: Can we add a short name to this crd?

File: helm/charts/hpe-csi-driver/crds/hpe-replicated-device-info-crd.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                  properties:
+                    targetName: 
+                      description: Target Name of the array
+                      type: string
+                    targetCpg:
+                      description: Target CPG of the array
+                      type: string
+                    targetSnapCpg:
+                      description: Target Snap CPG of the array
+                      type: string
```
Comment (sneharai4) [inline helm/charts/hpe-csi-driver/crds/hpe-replicated-device-info-crd.yaml:32]: Can this be made as optional param for the targets?

File: helm/charts/hpe-csi-driver/crds/hpe-replicated-device-info-crd.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  creationTimestamp: null
+  name: hpereplicationdeviceinfos.storage.hpe.com
+spec:
+  group: storage.hpe.com
+  names:
+    kind: HPEReplicationDeviceInfo
+    plural: hpereplicationdeviceinfos
```
Comment (sneharai4) [inline helm/charts/hpe-csi-driver/crds/hpe-replicated-device-info-crd.yaml:10]: shortNames:
    - hperdi

Can be added here after plural.

File: helm/charts/hpe-csi-driver/crds/hpe-replicated-device-info-crd.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+                      type: string
+                    targetCpg:
+                      description: Target CPG of the array
+                      type: string
+                    targetSnapCpg:
+                      description: Target Snap CPG of the array
+                      type: string
+                    targetSecret:
+                      description: Secret of the replicated array 
+                      type: string
```
Comment (rgcostea) [inline helm/charts/hpe-csi-driver/crds/hpe-replicated-device-info-crd.yaml:38]: What about targetSecretNamespace?  Without it, we won't know which namespace in which to look up the secret.

Comment (sneharai4) [reply helm/charts/hpe-csi-driver/crds/hpe-replicated-device-info-crd.yaml:38]: We do a lookup on all the namespaces and get the secret details.

**Review by sneharai4 — APPROVED** [review]

File: helm/charts/hpe-csi-driver/crds/hpe-replicated-device-info-crd.yaml (hunk 4 above)
Comment (rgcostea) [reply helm/charts/hpe-csi-driver/crds/hpe-replicated-device-info-crd.yaml:38]: What if you have two secrets with the same name in 2 different namespaces?

Your lookup would presumably get the first one but that might not be the one the user wants.

Comment (wdurairaj) [reply helm/charts/hpe-csi-driver/crds/hpe-replicated-device-info-crd.yaml:38]: we are adding `targetSecretNamespace:` in the CRD , and the CSP will read only the secret from this namespace. Will make the CSP for that.

**Pull request merged**